scvi/dataset/synthetic.py

```diff
@@ -42,3 +42,118 @@ def __init__(self, mu=4.0, theta=2.0, dropout=0.7, save_path='data/'):
+
+        assert mode in ['mixed', 'zi', 'nb'], 'Mode {} not recognized'.format(mode)
+        np.random.seed(0)
+
+        if n_genes_total % n_clusters > 0:
+            print("Warning, clusters have inequal sizes")
+
+        if n_genes_high > (n_genes_total // n_clusters):
+            print("Overlap of", n_genes_high - (n_genes_total // n_clusters), "genes")
+
+        # Generate data before dropout
+        batch_size = n_cells_cluster * n_clusters
+        expression_mat = np.ones((n_batches, batch_size, n_genes_total))
+
+        self.batch_size = batch_size
+        self.n_batches = n_batches
+        self.n_genes_total = n_genes_total
+
+        labels = np.ones((n_batches, batch_size, 1))
+
+        # For each cell cluster, some genes have a high expression, the rest
+        # has a low expression. The scope of high expression genes "moves"
+        # with the cluster
+        for cluster in range(n_clusters):
+            ind_first_gene_cluster = cluster * (n_genes_total // n_clusters)
+            ind_last_high_gene_cluster = ind_first_gene_cluster + n_genes_high
+
+            # Weights in a cluster to create highly-expressed and low-expressed genes
+            weights = weight_low * np.ones((n_genes_total,))
+            weights[ind_first_gene_cluster:ind_last_high_gene_cluster] = weight_high
+            weights /= weights.sum()
+
+            exprs = np.random.poisson(lam_0 * weights, size=(n_batches, n_cells_cluster,
+                                                             n_genes_total))
+            expression_mat[:, cluster * n_cells_cluster:(cluster + 1) * n_cells_cluster, :] = exprs
+            labels[:, cluster * n_cells_cluster:(cluster + 1) * n_cells_cluster, :] = cluster
+
+        # Apply dropout depending on the mode
+
+        new_data = self.mask(expression_mat)
+
+        super().__init__(
+            *GeneExpressionDataset.get_attributes_from_list(new_data, list_labels=labels),
+            gene_names=np.arange(n_genes_total).astype(np.str))
+
+    def mask(self, data):
+        return data
+
+
+class ZISyntheticDatasetCorr(SyntheticDatasetCorr):
+    def __init__(self, dropout_coef=1.0, lam_dropout=1.0, **kwargs):
+        self.dropout_coef = dropout_coef
+        self.lam_dropout = lam_dropout
+        super(ZISyntheticDatasetCorr, self).__init__(**kwargs)
+
+    def mask(self, data):
+        p_dropout = self.dropout_coef * np.exp(-self.lam_dropout * (data**2))
+        # Probability of failure
+        mask = np.random.binomial(n=1, p=1 - p_dropout,
+                                  size=(self.n_batches, self.batch_size, self.n_genes_total))
+        return data * mask
+
+
+# class MixedSyntheticDatasetCorr(SyntheticDatasetCorr):
+#     def __init__(self, p_dropout=0.2, ratio_genes_zi=0.5, **kwargs):
+#         self.p_dropout = p_dropout
+#         self.ratio_genes_zi = ratio_genes_zi
+#         self.zi_genes_idx = None
+#         super(MixedSyntheticDatasetCorr, self).__init__(**kwargs)
+#
```

Comment:
Better not to commit commented out code: https://kentcdodds.com/blog/please-dont-commit-commented-out-code

Reply:
Sorry about that, we wanted to keep a trace of that piece of code since it was not committed before ... We should not have done that!
Thanks for the advice!